File: 프로그래머스/2/42885. 구명보트/구명보트.java
import java.util.*;
class Solution {
     public int solution(int[] people, int limit) {
        Arrays.sort(people);
        ArrayDeque<Integer> deque = new ArrayDeque<>();
        for (int person : people) {
            deque.add(person);
        }
        int count = 0 ;
        while (deque.size()>1) {
            Integer last  = deque.pollLast();
            Integer first  = deque.pollFirst();
            if(first+last > limit){
                count++ ;
                deque.addFirst(first);
            } else if (first+last <= limit) {
                count++;
            }
        }
        if(deque.size() == 1 ){
            count++ ; 
        }
        return count;
    }
}

File: 프로그래머스/2/42885. 구명보트/구명보트.js
function solution(people, limit) {
  let biggest = 0,
    count = 0,
    i = 0;
  people.sort((a, b) => a - b);
  while (people.length > 0) {
    biggest = people.pop();
    i = 0;
    while (people[i] <= limit - biggest) i++;
    if (i) people.splice(i - 1, 1);
    count++;
  }
  return count;
}